tests/Skills/Parent/SkillsHelper.cs

```diff
@@ -0,0 +1,130 @@
+                    var oauthCard = ((JObject)attachment.Content).ToObject<OAuthCard>();
+
+                    if (oauthCard.TokenExchangeResource != null)
+                    {
+                        try
+                        {
+                            // AAD token exchange
+                            var result = await _tokenExchangeProvider.ExchangeTokenAsync(
+                                turnContext,
+                                _parentConnectionName,
+                                activity.Recipient.Id,
+                                new TokenExchangeRequest() { Uri = oauthCard.TokenExchangeResource.Uri }).ConfigureAwait(false);
+
+                            if (!string.IsNullOrEmpty(result.Token))
+                            {
+                                // Send an Invoke back to the Skill
+                                return await SendTokenExchangeInvokeToSkill(turnContext, activity, result.Token, oauthCard.ConnectionName, default(CancellationToken)).ConfigureAwait(false);
+                            }
+                        }
+                        catch
+                        {
+                            return false;
+                        }
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private async Task<bool> SendTokenExchangeInvokeToSkill(ITurnContext turnContext, Activity incomingActivity, string token, string connectionName, CancellationToken cancellationToken)
+        {
+            var activity = incomingActivity.CreateReply() as Activity;
+            activity.Type = ActivityTypes.Invoke;
+            activity.Name = SignInConstants.TokenExchangeOperationName;
+            activity.Value = new TokenExchangeInvokeRequest()
+            {
+                Token = token,
+                ConnectionName = connectionName,
+            };
+
+            var skillConversationReference = await _conversationIdFactory.GetSkillConversationReferenceAsync(incomingActivity.Conversation.Id, cancellationToken).ConfigureAwait(false);
+            activity.Conversation = skillConversationReference.ConversationReference.Conversation;
+            activity.ServiceUrl = skillConversationReference.ConversationReference.ServiceUrl;
+
+            // route the activity to the skill
+            var response = await PostActivityAsync(activity, cancellationToken);
+
+            // Check response status: true if success, false if failure
+            var success = IsSucessStatusCode(response.Status);
+            if (success)
+            {
+                await turnContext.SendActivityAsync("token exchange successful");
+            }
+            else
+            {
+                await turnContext.SendActivityAsync("token exchange failed");
+            }
+
+            return success;
+        }
+
+        private bool IsSucessStatusCode(int statusCode)
+        {
+            return statusCode >= 200 && statusCode <= 299;
+        }
```

Comment:
This is only used in one spot, I don't think it needs to be a helper function...